models/eprop_synapse.h

```diff
@@ -0,0 +1,585 @@
+Wmax       pA    :math:`W_{ji}^\text{max}`    100.0  Maximal value for synaptic weight
+Wmin       pA    :math:`W_{ji}^\text{min}`      0.0  Minimal value for synaptic weight
+=========  ====  =========================  =======  ===============================================================
+
+Recordables
++++++++++++
+
+The following variables can be recorded.
+
+  - ``weight``
+
+Usage
++++++
+
+This model can only be used in combination with the other e-prop models,
+whereby the network architecture requires specific wiring, input, and output.
+The usage is demonstrated in a
+:doc:`supervised regression task <../auto_examples/eprop_plasticity/eprop_supervised_regression/>`
+and a :doc:`supervised classification task <../auto_examples/eprop_plasticity/eprop_supervised_classification>`,
+reproducing the original proof-of-concept tasks in [1]_.
+
+Transmits
++++++++++
+
+SpikeEvent, DSSpikeEvent
+
+References
+++++++++++
+
+.. [1] Bellec G, Scherr F, Subramoney F, Hajek E, Salaj D, Legenstein R,
+       Maass W (2020). A solution to the learning dilemma for recurrent
+       networks of spiking neurons. Nature Communications, 11:3625.
+       https://doi.org/10.1038/s41467-020-17236-y
+
+.. [2] Kingma DP, Ba JL (2015). Adam: A method for stochastic optimization.
+       Proceedings of International Conference on Learning Representations (ICLR).
+       https://doi.org/10.48550/arXiv.1412.6980
+
+.. [3] Korcsak-Gorzo A, Stapmanns J, Espinoza Valverde JA, Dahmen D,
+       van Albada SJ, Bolten M, Diesmann M. Event-based implementation of
+       eligibility propagation (in preparation)
+
+See also
+++++++++
+
+EndUserDocs */
+
+class EpropCommonProperties : public CommonSynapseProperties
+{
+public:
+  EpropCommonProperties();
+
+  void get_status( DictionaryDatum& d ) const;
+  void set_status( const DictionaryDatum& d, ConnectorModel& cm );
+
+  bool adam_;
+  double adam_beta1_;
+  double adam_beta2_;
+  double adam_epsilon_;
+  long batch_size_;
+  double recall_duration_;
+};
+
+EpropCommonProperties::EpropCommonProperties()
+  : CommonSynapseProperties()
+  , adam_( false )
+  , adam_beta1_( 0.9 )
+  , adam_beta2_( 0.999 )
+  , adam_epsilon_( 1e-8 )
+  , batch_size_( 1 )
+  , recall_duration_( 1.0 )
+{
+}
+
+void
+EpropCommonProperties::get_status( DictionaryDatum& d ) const
+{
+  CommonSynapseProperties::get_status( d );
+  def< bool >( d, names::adam, adam_ );
+  def< double >( d, names::adam_beta1, adam_beta1_ );
+  def< double >( d, names::adam_beta2, adam_beta2_ );
+  def< double >( d, names::adam_epsilon, adam_epsilon_ );
+  def< long >( d, names::batch_size, batch_size_ );
+  def< double >( d, names::recall_duration, recall_duration_ );
+}
+
+void
+EpropCommonProperties::set_status( const DictionaryDatum& d, ConnectorModel& cm )
+{
+  CommonSynapseProperties::set_status( d, cm );
+  updateValue< bool >( d, names::adam, adam_ );
+  updateValue< double >( d, names::adam_beta1, adam_beta1_ );
```

Comment:
Add checks for parameter validity.

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/2867/commits/806d78172a380585a625ed901826484674fc1997